services/QuillLMS/app/services/serialize_vitally_sales_account.rb

```diff
@@ -49,25 +54,16 @@ def data
     }
   end
 
-  private def activities_per_student(active_students, activities_finished)
-    if active_students > 0
-      (activities_finished.to_f / active_students).round(2)
+  private def get_from_cache(key)
+    last_school_year = Date.today.year - 1
+    cached_data = $redis.get("school_id:#{@school.id}_vitally_stats_for_year_#{last_school_year}")
+    if cached_data.present?
+      parsed_data = JSON.parse(cached_data)
     else
-      0
+      PreviousYearSchoolDatum.new(@school, last_school_year).calculate_and_save_data
+      parsed_data = JSON.parse($redis.get("school_id:#{@school.id}_vitally_stats_for_year_#{last_school_year}") || '{}')
     end
-  end
-
-  private def active_students_query(school)
```

Comment:
I think you can use an [unscoped block](https://github.com/rails/rails/issues/13775#issue-25947654) and avoid the Raw SQL.